Kalman BMTF Emu v2.7

panoskatsoulis opened the pull request:
This is the latest BMTF emulator for the Kalman algorithm

kandrosov committed:
First implementation of deep tau id.

ocolegrove committed:
Building dpf isolation module

ocolegrove committed:
Adding in v1

ocolegrove committed:
Adding in runTauIDMVA for other users

ocolegrove committed:
making things fully reproducible

mbluj committed:
Reorganisation of configuration files: cff split to cfi and cff

mbluj committed:
Some code cleaning

mbluj committed:
adapt to cfi/cff reorganization

bachtis committed:
New version

panoskatsoulis committed:
spare checked-out the remaining differences from bachtis/KMTF_V2.7.1

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/6819

cmsbuild commented:
A new Pull Request was created by  @panoskatsoulis  for master.

It involves the following packages:

DataFormats/L1TMuon
L1Trigger/L1TMuonBarrel


@nsmith-, @rekovic, @cmsbuild, @thomreis can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @rovere, @thomreis this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 13f5b189d82a1f65fe7ccdd55402cfbfdfb75787

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31016/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation warning when building:  See details on the summary page.

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

thomreis commented:
Hi @panoskatsoulis 
since the master points to the 10_4 branch now please also make a backport PR to 10_3_X after fixing the compilation warnings.

thomreis commented:
Hi @panoskatsoulis 
please fix the compilation warnings in order to go ahead with this PR's integration.

panoskatsoulis committed:
Clang 7.0.0 Compilation Warning Fix

panoskatsoulis commented:
hi @thomreis 
the procedure fails because of 2 Clang warnings (namely **-Wself-assign-overloaded** & **-Wheader-guard**).
The last commit I've pushed fixes them in my local tests.

In details, -Wheader-guard is just a typo in the first line of the file
'L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelKalmanLUTs.h',
but -Wself-assign-overloaded raised during compilation because of the line 226 in the file
'L1Trigger/L1TMuonBarrel/src/L1TMuonBarrelKalmanRegionModule.cc'.
**line 226:   s3_3=s3_3;**
The assignment of an object to itself now raises this warning.
In the release notes of the Clang ([link](http://ftp.netbsd.org/pub/NetBSD/NetBSD-current/src/external/bsd/llvm/dist/clang/docs/ReleaseNotes.rst)) is proposed to cast the rhs as reference of the object, in order to suppress the warning.
**line 226:   s3_3=static_cast<L1MuKBMTrack&>(s3_3);**

Can we please test again and see if this fixes the issue?
Also, @bachtis confirm please that this fix will not affect the algorithm.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/6854

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

thomreis commented:
Hi @panoskatsoulis 
what is the point of assigning an object to itself? I can see the point of having this in FW but not in an emulator.

panoskatsoulis commented:
I would say possibly for following the same steps exactly as in FW (the difference of the emulation concept wrt the simulation concept).
I think there is a reason why @bachtis is doing this.

thomreis commented:
OK, so the @bachtis can confirm if this is still compatible with the FW.
In the meantime we can already test this.
Please also update the 10_2 backport and create a 10_3_X one when the fix is confirmed.

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

panoskatsoulis commented:
sure, I'll update the other PR while we wait for answer and the test results

bachtis commented:
Guys wait. Maybe CLANG found a bug here . Looking to it

bachtis commented:
OK looked at it. @panoskatsoulis  please replace the line 226 to      
s3_3=s2_3; 
and update the PR .
In short: This code is a mimicing of the hardware sorter  in firmware so it is more complicated than a std::sort for data/MC agreement to work . The line of self assesment is a typo .

panoskatsoulis committed:
Changed the typo bug that was causing -Wself-assign-overloaded

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/6867

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

thomreis commented:
please test

panoskatsoulis commented:
good, the typo has been corrected.
Although it wont correct the RelVal issues I think.
@thomreis please wait for the test, I need to check the logs

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d19015ceec631b75bd0737435684d619e920f648
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31090/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31090/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2994843
 - DQMHistoTests: Total failures: 52
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2994594
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 134 log files, 14 edm output root files, 32 DQM output files

thomreis commented:
Hi @panoskatsoulis 
are the failing histograms expected with this emulator change?

thomreis commented:
Please provide a PR to 10_3_X as well.

panoskatsoulis commented:
Hi all,
The failing plots belong all in the "Secondary" L1TEMU/L1TdeStage2BMTF folder.
For the run 297557 (is used by the workflow 136.788) in this folder are the Kalman Algorithm Validation plots.
However, the firmware which is used for this run doesn't include any Kalman Algorithm at all.
The Unpacker in this case creates a null collection that is being compared to the Kalman output giving a validation that is completely without sense.

Maybe I need to understand what are the data points that lay on these plots in such cases (because they might be leftovers in the memory) but this is completely irrelevant to the validity of this PR.

The failing plots are not related with the PR'ed Emulator.

thomreis commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

panoskatsoulis commented:
hold

panoskatsoulis committed:
Optimization for data/MC agreement of the TrackAddress map

panoskatsoulis commented:
unhold

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/6907

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

MRD2F committed:
Review of DPF and DeepTauId code.

- Defined base class for deep tau discriminators.
- Removed weight files from home cms repository. Now using weights from cms-data.
- Defined WP for both discriminators. Now all discriminators return the corresponding WP results.
- Removed cfi files. Using fillDescriptions instead.
- General code review and cleaning.

cmsbuild commented:
+1
Tested at: f3e2b3f4c15d84a342d5b7c128100231b669785f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31151/summary.html

cmsbuild commented:
Comparison job queued.

MRD2F committed:
Added example of a python configuration file to produce pat::Tau collection with the new Tau-Ids

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31151/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 2994843
 - DQMHistoTests: Total failures: 67
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2994579
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 134 log files, 14 edm output root files, 32 DQM output files

MRD2F committed:
requested changes on runDeepTauIDsOnMiniAOD.py

mbluj committed:
Clean runTauIdMVA.py tool and test config to run tauIDs

MRD2F committed:
Fixed problems to be compatible with branch CMSSW_10_4_X

thomreis commented:
Hi @panoskatsoulis 
are the failing comparisons expected?

panoskatsoulis commented:
Hi @thomreis 
which are failing comparisons? Can't find them

MRD2F committed:
Made DeepTauId and DPFIsolation thread-safe

thomreis commented:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_4_X_2018-10-18-2300+24855/28994/136.788_RunSinglePh2017B+RunSinglePh2017B+HLTDR2_2017+RECODR2_2017reHLT_skimSinglePh_Prompt+HARVEST2017/L1TEMU_L1TdeStage2BMTF_L1TdeStage2BMTF-Secondary.html for example but there could be others.

MRD2F committed:
Finish implement thread-safe requirements on DPFIsolation

MRD2F committed:
Remove trace of TauIdMVAAuxiliaries in DeepTauId

panoskatsoulis commented:
@thomreis 
I've checked the failures. I would say they are not problem.
The test is checking consistency of the master release with and without the PR for the emulator plots.
The emulator plots without the PR are wrong so this is why we pushed this. The latest (PR version) emulator and the deprecated (master version) emulator is expected that they won't agree in their mismatches.

There is no way to have consistency of the mismatches with the new and the old Emulator I think.
The failures, all, seem to come from the "Secondary" folder where the Kalman comparison is taking place. Did you find any different?

thomreis commented:
+1

@panoskatsoulis thanks for checking.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

MRD2F committed:
Disable DPFTau_2016_v1 and issue some warnings

mbluj committed:
Merge pull request #100 from MRD2F/CMSSW_10_4_X_DPFIsoDeepTau

DNN tau IDs for CMSSW_10_4_X

MRD2F committed:
- Implemented on runTauIdMVA the option to work with new training files quantized
- Added a new parameter 'version' on runTauIdMVA, used on DPFIsolation
- Changes on DeepTauId to reduce memory consumption

bachtis commented on L1Trigger/L1TMuonBarrel/test/kalmanTools/makeEtaLuts.py:
```diff
@@ -1,4 +1,3 @@
-from __future__ import print_function
```
I didnt add this piece in the code. probably someone did it to make it compile in different python version. Just delete the line

fabiocos commented:
@panoskatsoulis @bachtis all the python fragments must be python3 compatible, and @davidlange6 made a CMSSW-wide migration for this a few months ago. So deleting that line is not an option, you should just move the syntax of the print statement to the new one. It should be a fairly straightforward adjustment.

fabiocos commented:
hold

pending the required fixes

cmsbuild commented:
Pull request has been put on hold by  @fabiocos
They need to issue an `unhold` command to remove the `hold` state or L1 can `unhold` it for all

MRD2F committed:
- Implementation of global cache to avoid reloading graph for each thread and reduce the memory consuption
- Creation of class DeepTauCache in DeepTauBase, in which now is created graph and session
- Implementation of two new static methods inside the class DeepTauBase: initializeGlobalCache and globalEndJob. The graph and DeepTauCache object are created now inside initializeGlobalCache

kandrosov committed:
Implemented TauWPThreshold class.

TauWPThreshold class parses WP cut string (or value) provided in the
python configuration. It is needed because the use of the standard
StringObjectFunction class to parse complex expression results in an
extensive memory usage (> 100 MB per expression).

MRD2F committed:
Applied changes on DeepTauBase to allow load new training files using memory mapping

MRD2F committed:
Remove the qm.pb input files and leaving just the quantized and the original files

MRD2F committed:
-Overall, changes to improve memory usage, among these are:
	- Implementation of global cache to avoid reloading graph for each thread
	- Creation of two new static methods inside the class DeepTauBase: initializeGlobalCache and globalEndJob. The graph and DeepTauCache object are created now inside initializeGlobalCache. The memory consumption of initializeGlobalCache for the original, quantized and files that are load using memory mapping method are in the memory_usage.pdf file
	- Implemented configuration to use new training files quantized, and set them as default
	- Implementation of configuration for load files using memory mapping. In our case there wasn't any improvement, respect at the memory consumption of this method, respect the quantized files, so this is not used, but set for future training files
- General code review and cleaning.

MRD2F committed:
Applied style comments

fabiocos commented:
@panoskatsoulis any news? the requested update looks trivial to be implemented

MRD2F committed:
Applied style comments

MRD2F committed:
Applied comments

MRD2F committed:
Change to be by default the original training file for deepTau, instead of the quantized

MRD2F committed:
Applied commets of previus PR

MRD2F committed:
cleaning code

MRD2F committed:
Modification in the config to work with new label in files

MRD2F committed:
Applied comment about the expected format of name of training file

MRD2F committed:
Fix in last commit

MRD2F committed:
Applied last comments

MRD2F committed:
Changes regarding forward-porting DNN-related developments from the PRs #105 and #106 from 94X to 104X

mbluj committed:
Merge pull request #107 from MRD2F/CMSSW_10_4_0_pre2_DNN

forward-porting DNN-related from branch cms-tau-pog:CMSSW_9_4_X_tau_pog_DNNTauIDs

mbluj committed:
Remove assigning value of variable to itself

MRD2F committed:
Applied @perrotta comments on 104X

MRD2F committed:
Fix error

MRD2F committed:
Applied comments

MRD2F committed:
Applied comments

MRD2F committed:
Merge remote-tracking branch 'tauPOG/CMSSW_10_4_X_tau_pog_DNNTauIDs' into CMSSW_10_4_0_pre2_DNN

MRD2F committed:
Fix merge problem

MRD2F committed:
Applied a few commets

MRD2F committed:
Applied more changes

MRD2F committed:
Applied a few small followups

MRD2F committed:
Fixed error on DPFIsolation

mbluj committed:
Merge pull request #109 from MRD2F/CMSSW_10_4_0_pre2_DNN

Applied comments

mbluj committed:
Update DPFIsolation.cc

panoskatsoulis commented:
Hi @thomreis , @fabiocos , sorry for the delay... I will update it in a few days.

panoskatsoulis committed:
redoing the migration to python3

panoskatsoulis committed:
changed 'python *' to python(*)

cmsbuild commented:
The code-checks are being triggered in jenkins.

panoskatsoulis commented:
Hi, again @thomreis, @fabiocos 
i've redone the python3 migration, and also changed all print statements to the python3 print() function

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/7365

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

panoskatsoulis commented:
@fabiocos,
will it be needed to make the same change to PR #24934 as well?
That one is related to 103X.

cmsbuild commented:
+1
Tested at: 069f23991de3bba7a7620c2f680322a08506d19c
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31849/summary.html

cmsbuild commented:
Comparison job queued.

thomreis commented:
Hi @panoskatsoulis 
yes.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31849/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3131839
 - DQMHistoTests: Total failures: 61
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3131574
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

fabiocos commented:
@thomreis if the backport to 10_3_X is really needed (is it now at the end of HI data taking ?) the code should be updated accordingly.

thomreis commented:
Hi @panoskatsoulis can you check if all the failing comparisons are expected:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_4_X_2018-11-25-2300+24855/29530/validateJR.html
Please take a look also at the static checks https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31849/llvm-analysis/

panoskatsoulis committed:
resolving the 'Dead Assignment' Clang issue

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/7459

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

panoskatsoulis commented:
Hi @thomreis 
after talking with @bachtis, It seems that some lines from the src were of no need.
I've updated the PR once again.

thomreis commented:
please test

If the lines are not needed it would be better to remove them completely.

cmsbuild commented:
The tests are being triggered in jenkins.

panoskatsoulis commented:
It is possible the commented out lines to be useful under debugging circumstances. I wouldn't remove them without @bachtis 's input on the issue.

cmsbuild commented:
+1
Tested at: 1d793016be4181f82abd937cc2f8c780b958d1c4
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/31984/summary.html

cmsbuild commented:
Comparison job queued.

thomreis commented:
Hi @panoskatsoulis 
are the differences in the test WFs expected?

cmsbuild committed:
Merge pull request #25016 from cms-tau-pog/CMSSW_10_4_X_tau_pog_DNNTauIDs

DNN-based Tau-Id discrimians

cmsbuild committed:
Merge pull request #25360 from DryRun/CMSSW_10_4_0_pre2_hcaldqm

HCALDQM: Move channel quality lookup to dqmBeginLuminosityBlock (+increase pedestal RMS axis)

cmsbuild committed:
Merge pull request #25382 from vargasa/GeoCompAirCtg

Validation/Geometry: Integrate geometry comparison to MaterialBudget.py

cmsbuild committed:
Merge pull request #25390 from ColeLindsey/vertexSmearingPbPb

Added 2018 PbPb collision vertex smearing parameters

cmsbuild committed:
Merge pull request #25430 from Dr15Jones/clangL1TriggerL1TMuonBarrel

Fixed broken include guard in L1TMuonBarrelKalmanLUTs

cmsbuild committed:
Merge pull request #25418 from gartung/FWCore-Services-StallMonitor-micro

FWCore: Update StallMonitor service to use microsecond timing

davidlange6 commented:
Your PR is unmergeable. Please have a look and possibly rebase it.

cmsbuild committed:
Merge pull request #25423 from peruzzim/fixNanoValidation

Remove workflow 1325.9 testing NanoAOD on 92X

cmsbuild committed:
Merge pull request #25417 from fabiocos/fc-mtdTopology

MTD geometry: move topology parameters from PSet to xml, adapt the code

cmsbuild committed:
Merge pull request #25426 from igv4321/charge-asymmetry-switch

Switching the mode of charge asymmetry evaluation

cmsbuild committed:
Merge pull request #25359 from ianna/geom-ctpps-detgeom-cleanup

RECO Geometry: CTPPS DetGeomDesc Cleanup

cmsbuild committed:
Merge pull request #25422 from tocheng/104X_OfflineECALhistory

[104X] Change GT. Offline GT Fix ECAL Queues

cmsbuild committed:
Merge pull request #25427 from slomeo/Cavern2018Fix

Bugfix for cavern

cmsbuild committed:
Merge pull request #25429 from Dr15Jones/clangL1TriggerL1THGCal

Fixed clang warnings in L1Trigger/L1THGCal

peruzzim committed:
Define new run2_miniAOD_devel and run2_nanoAOD_102Xv1 modifiers

peruzzim committed:
Add modifiers to supported list

Dr15Jones committed:
Only call consumes in ElectronSeedProducer if actually needed

A data product is only consumed if a particular parameter value
is used.

cmsbuild committed:
Merge pull request #25438 from civanch/extra_physlists

Added extra Physics Lists for neutron background study

cmsbuild committed:
Merge pull request #25397 from vargasa/RefPlot

Validation/Geometry: Add reference plots to Material Budget test

cmsbuild committed:
Merge pull request #25364 from wddgit/returnUniquePtrFromFetchPayload

Return unique_ptr from fetchPayload

makortel committed:
Remove unused functions

cmsbuild committed:
Merge pull request #25451 from makortel/fixGeometryGlobalTrackingGeometryBuilderBuildFile

Remove duplicate entry in Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml

cmsbuild committed:
Merge pull request #25449 from Dr15Jones/consumesElectronSeedProducer

Only call consumes in ElectronSeedProducer if actually needed

cmsbuild committed:
Merge pull request #25454 from makortel/fixClangTrackingToolsPatternTools

Fix clang warnings in TrackingTools/PatternTools

cmsbuild committed:
Merge pull request #25450 from Dr15Jones/clangMuonAnalysisMomentumScaleCalibrartion

fix clang warnings in MuonAnalysis/MomentumScaleCalibration

cmsbuild committed:
Merge pull request #25443 from peruzzim/addMiniNanoModsPostAutumn18

Define new run2_miniAOD_devel and run2_nanoAOD_102Xv1 modifiers

cmsbuild committed:
Merge pull request #25401 from civanch/sim_magnetic_field

Sim magnetic field

panoskatsoulis commented:
The following latest changes are the problem, I will update it again to be compatible with the 10_4_X interface.

<<<<<<< BMTF_KMTF-Emu-v2.7_forMaster
#ifndef __L1TMuonBarrelKalmanLUTs_h
#define __L1TMuonBarrelKalmanLUTs_h
=======
#ifndef L1Trigger_L1TMuonBarrel_L1TMuonBarrelKalmanLUTs_h
#define L1Trigger_L1TMuonBarrel_L1TMuonBarrelKalmanLUTs_h
>>>>>>> master

panoskatsoulis commented:
Hi @thomreis
the failures seem to be because of the old runs which have been used in the workflows, and the non-well-defined behavior of the DQM AlgoSelector plugin under these runs.
There is a default behavior of the "algoselector" to prevent segfaults when is processing runs in which there was no Kalman firmware running at the time.
However, this plugin has been working correctly with the latest runs at P5 for more than 2 months and also is not part of this PR.

panoskatsoulis committed:
Merge branch 'master' into BMTF_KMTF-Emu-v2.7_forMaster

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-24855/7543

cmsbuild commented:
Pull request #24855 was updated. @nsmith-, @rekovic, @cmsbuild, @thomreis can you please check and sign again.

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: a11ff066a49ca481c4638fe95284f9f32c9fe075
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32091/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/a11ff066a49ca481c4638fe95284f9f32c9fe075
https://github.com/cms-sw/cmssw/commit/772f26f37bb6603475a676cca4fa7161efb95cfd
https://github.com/cms-sw/cmssw/commit/493867bbaf068ba0d0c713ae22359d0e02f232f5
https://github.com/cms-sw/cmssw/commit/677b794d87554c9e74d10edb3e0c2598857f3607
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32091/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32091/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32091/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3136422
 - DQMHistoTests: Total failures: 61
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3136157
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

panoskatsoulis committed:
Merge branch 'master' into BMTF_KMTF-Emu-v2.7_forMaster

thomreis commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 75642cf79071312c53250c7c323b24c4fb52599d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32121/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/75642cf79071312c53250c7c323b24c4fb52599d
https://github.com/cms-sw/cmssw/commit/3fb064722ffba8ec392f8c654eb01ecb6f7022bd
https://github.com/cms-sw/cmssw/commit/e92ecf1a89f0a188dfe5e799cc223f64ac52f91a
https://github.com/cms-sw/cmssw/commit/5281ba1e17695d6a1b70b5b672858a589a654e11
https://github.com/cms-sw/cmssw/commit/d72e88219adfb4e4887e991274b638a22133799a
https://github.com/cms-sw/cmssw/commit/6cd52ac8976828ae9c4a11c9bd8fa0925420513e
https://github.com/cms-sw/cmssw/commit/23c90f4745a429cdd8a8a33af7a7ee519026d200
https://github.com/cms-sw/cmssw/commit/2df2335302456b156c2775d453b6571d58c5e346
https://github.com/cms-sw/cmssw/commit/fffd49634f224bddb50ea300a8a298e752b5ac01
https://github.com/cms-sw/cmssw/commit/77915b61da98aee9a50d5f7158521d6f079537d3
https://github.com/cms-sw/cmssw/commit/a11ff066a49ca481c4638fe95284f9f32c9fe075
https://github.com/cms-sw/cmssw/commit/772f26f37bb6603475a676cca4fa7161efb95cfd
https://github.com/cms-sw/cmssw/commit/493867bbaf068ba0d0c713ae22359d0e02f232f5
https://github.com/cms-sw/cmssw/commit/677b794d87554c9e74d10edb3e0c2598857f3607
https://github.com/cms-sw/cmssw/commit/ed3bfda4c2beaadff197df621013099f53477f37
https://github.com/cms-sw/cmssw/commit/57859900e5a81f6a92a31a179c3aaeb7add38500
https://github.com/cms-sw/cmssw/commit/5b6132361faec981590336652ffa44759a146074
https://github.com/cms-sw/cmssw/commit/39496c065ac2310955297b8f7aa194c9b3f59350
https://github.com/cms-sw/cmssw/commit/dd5d9186f2ede0f7ae83d741aec9097c23a6bc77
https://github.com/cms-sw/cmssw/commit/5b3b317fdfb9a869076f564ec49ac7701ff3d08d
https://github.com/cms-sw/cmssw/commit/2d4e11aee519678dc59762c20f08b59c7ed483f8
https://github.com/cms-sw/cmssw/commit/14d4e71448b0d7beb5ab750ee05c297094f550ef
https://github.com/cms-sw/cmssw/commit/500def780a8b3164d98e2a505dc9b208ede10900
https://github.com/cms-sw/cmssw/commit/f7cf8dc05c7702c985af0ab1a20da77c56cd308c
https://github.com/cms-sw/cmssw/commit/380f1a4c81026f8223d75385dcb8a2699fe0e626
https://github.com/cms-sw/cmssw/commit/27bbf6e74e47b1f16073ebed67266915b97a2238
https://github.com/cms-sw/cmssw/commit/60a254cf279cde87541bef91a60298cd50743b20
https://github.com/cms-sw/cmssw/commit/b7159179438d9cd6a6431490ad8bd0a93488fb7a
https://github.com/cms-sw/cmssw/commit/f246f43be0208ee5608782b623500dfba6985a24
https://github.com/cms-sw/cmssw/commit/8b204e45b9b0ba365049c0ac1191e0eacf8d3e4d
https://github.com/cms-sw/cmssw/commit/8faf9797c5c7290f03f31bce3f09cbef592c4fab
https://github.com/cms-sw/cmssw/commit/6c740721568fede87719449b94e11e900cf2cbe1
https://github.com/cms-sw/cmssw/commit/7a65e90a0c7ef76236b19f5532bbbe00984a0b1e
https://github.com/cms-sw/cmssw/commit/61321279ee46f369212992e86e9d6f9af46e48c9
https://github.com/cms-sw/cmssw/commit/a27ec47d770bb2aaef8ffb873982f5b43da49e77
https://github.com/cms-sw/cmssw/commit/1f7811388d37d03f078e6da39fa54eb88d438478
https://github.com/cms-sw/cmssw/commit/9173af878af87638210fd7e21f8cd65c8190ceb7
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32121/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32121/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32121/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3136422
 - DQMHistoTests: Total failures: 60
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3136158
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

thomreis commented:
Hi @panoskatsoulis 
there is a violation of CMS code rules https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32121/codeRules/
Please take a look.

thomreis commented:
In the meantime @fabiocos could you unhold this already please? The python fixes have been implemented already a while ago.

thomreis commented:
Hi @panoskatsoulis 
false alarm. The code rule violation must have been in a commit that was added to the tests but is not in this PR.
Please check the mismatches and confirm again that they are all expected.

fabiocos commented:
unhold

thomreis commented:
+1

The failing comparisons are the same than in previous tests and due to the use of old runs in the WFs where the Kalman FW was not active in data taking.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

panoskatsoulis commented:
Hi @thomreis, I've checked again. In deed the failures exist due to the old runs used by the workflow, and also we need to keep in mind that this PR changes the emulator. With that in mind, differences are being expected even for runs in which the emulator run.

thomreis commented:
Hi @panoskatsoulis thanks for the additional confirmation.
@fabiocos this is finally ready to be merged.

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-24855/32372/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3153717
 - DQMHistoTests: Total failures: 74
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3153439
 - DQMHistoTests: Total skipped: 204
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

fabiocos commented:
@thomreis @panoskatsoulis the PR looks ready for integration, just one clarification: I see several pieces of code that are not just debugging printouts commented: what is the purpose of that? In principle this might turn into a time bomb if not properly documented.

panoskatsoulis commented:
Hello, @fabiocos this question is for @bachtis .

bachtis commented:
Hi Fabio, We will clean it up some time in the future - but I suggest we merge this now. We will have additional developments and PRs in the near future so we can clean it up .

fabiocos commented:
+1
